src/isa/isainfo.h

```diff
@@ -47,28 +44,46 @@ struct RegInfoBase {
   virtual bool regIsReadOnly(unsigned i) const = 0;
 };
 
+/// An index into a single register.
+struct RegIndex {
+  const std::shared_ptr<const RegFileInfoInterface> file;
+  const unsigned index;
+};
+
+using RegInfoVec = std::vector<std::shared_ptr<const RegFileInfoInterface>>;
+using RegInfoMap =
+    std::map<std::string_view, std::shared_ptr<const RegFileInfoInterface>>;
+
 /// The ISAInfoBase class defines an interface for instruction set information.
 class ISAInfoBase {
 public:
   virtual ~ISAInfoBase(){};
   virtual QString name() const = 0;
   virtual ISA isaID() const = 0;
+  virtual const RegInfoMap &regInfoMap() const = 0;
 
-  std::optional<const RegInfoBase *>
-  regInfo(RegisterFileType regFileType = RegisterFileType::GPR) const {
-    if (auto match = m_regInfos.find(regFileType); match != m_regInfos.end()) {
-      return m_regInfos.at(regFileType).get();
-    } else {
-      return {};
+  const std::set<std::string_view> regFileNames() const {
```

Comment:
Why return a copied value as const?

Reply:
I think that's leftover from my plan of returning a const reference 😅